src/main/java/org/kiwiproject/dropwizard/util/health/MonitoredJobHealthCheck.java

```diff
@@ -0,0 +1,143 @@
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.kiwiproject.base.DefaultEnvironment;
+import org.kiwiproject.base.KiwiEnvironment;
+import org.kiwiproject.dropwizard.util.job.MonitoredJob;
+
+import java.time.Instant;
+
+@Slf4j
+@Getter(AccessLevel.PACKAGE) // For testing purposes
+public class MonitoredJobHealthCheck extends HealthCheck {
+
+    @VisibleForTesting
+    static final double DEFAULT_THRESHOLD_FACTOR = 2.0;
+
+    @VisibleForTesting
+    static final Duration MINIMUM_WARNING_THRESHOLD = Duration.minutes(1);
+
+    @VisibleForTesting
+    static final Duration DEFAULT_WARNING_DURATION = Duration.minutes(15);
+
+    private final MonitoredJob job;
+    private final Duration expectedFrequency;
+    private final Duration errorWarningDuration;
+    private final double thresholdFactor;
+    private final long lowerTimeBound;
+    private final KiwiEnvironment kiwiEnvironment;
+
+    @Builder
+    private MonitoredJobHealthCheck(MonitoredJob job,
+                                    Duration expectedFrequency,
+                                    Duration errorWarningDuration,
+                                    Double thresholdFactor,
+                                    Long lowerTimeBound,
+                                    KiwiEnvironment environment) {
+
+        this.job = requireNotNull(job, "job is required");
+        this.expectedFrequency = requireNotNull(expectedFrequency, "expectedFrequency is required");
+        this.errorWarningDuration = isNull(errorWarningDuration) ? DEFAULT_WARNING_DURATION : errorWarningDuration;
+        this.thresholdFactor = isNull(thresholdFactor) ? DEFAULT_THRESHOLD_FACTOR : thresholdFactor;
+        this.kiwiEnvironment = isNull(environment) ? new DefaultEnvironment() : environment;
+        this.lowerTimeBound = kiwiEnvironment.currentTimeMillis();
+    }
+
+    @Override
+    protected Result check() throws Exception {
+        try {
+            var lastRun = job.getLastSuccess().get();
+            if (!job.isActive()) {
+                return buildHealthyResult(f("Job is inactive. (last run: {})", instantToStringOrNever(lastRun)));
+            }
+
+            var now = kiwiEnvironment.currentTimeMillis();
+            var lastFailure = job.getLastFailure().get();
+            if ((now - lastFailure) < errorWarningDuration.toMilliseconds()) {
+                return buildUnhealthyResult(f("An error has occurred at: {}, which is within the threshold of: {}",
+                        instantToStringOrNever(job.getLastFailure().get()), errorWarningDuration));
+            }
+
+            var warningThreshold = getWarningThreshold();
+            if ((now - getTimeOrServerStart(lastRun)) > warningThreshold) {
+                return buildUnhealthyResult(f("Last successful execution was: {}, which is older than the threshold of: {} seconds",
+                        instantToStringOrNever(lastRun), Duration.milliseconds(warningThreshold).toSeconds()));
```

Comment:
Same question here: can `Duration.milliseconds(warningThreshold).toSeconds()` be calculated once at construction and stored in a (final) field?